include/boost/geometry/algorithms/detail/is_valid/polyhedral_surface.hpp

```diff
@@ -0,0 +1,453 @@
+// Boost.Geometry
+
+// Copyright (c) 2025 Oracle and/or its affiliates.
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_ALGORITHMS_DETAIL_IS_VALID_POLYHEDRAL_SURFACE_HPP
+#define BOOST_GEOMETRY_ALGORITHMS_DETAIL_IS_VALID_POLYHEDRAL_SURFACE_HPP
+
+#include <boost/qvm/vec.hpp>
+#include <boost/qvm/vec_operations.hpp>
+
+#include <boost/geometry/algorithms/detail/is_valid/interface.hpp>
+#include <boost/geometry/algorithms/detail/is_valid/polygon.hpp>
+#include <boost/geometry/algorithms/detail/calculate_point_order.hpp>
+#include <boost/geometry/algorithms/correct.hpp>
+#include <boost/geometry/io/wkt/wkt.hpp>
+#include <boost/geometry/strategies/transform/matrix_transformers.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail { namespace is_valid
+{
+
+template <typename PolyhedralSurface, bool AllowInconsistentOrientation = false>
+struct valid_intersection
+{
+    valid_intersection(size_t num_of_polygons)
+        : m_polygon_projections(num_of_polygons)
+    {}
+
+    using CT = typename coordinate_type<PolyhedralSurface>::type;
```

Comment:
Suggestion, `coor_t`? `CT` looks like a template parameter

Reply:
I agree, I changed it to `coordinate_type` which seems more common as a use in the library.